docs/usage/channels.rst

```diff
@@ -0,0 +1,452 @@
+
+.. glossary::
+
+    event
+        A single piece of data published to, or received from a :term:`backend`, bound
+        to the channel it was originally published to
+
+    event stream
+        A stream of :term:`events <event>`, consisting of events from all the channels a
+        :term:`Subscriber` has previously subscribed to
+
+    subscriber
+        A :class:`Subscriber <.subscriber.Subscriber>`: An object wrapping an
+        :term:`event stream` and providing access to it through various methods
+
+    backend
+        A :class:`ChannelsBackend <.backends.base.ChannelsBackend>`. This object
+        manages communication between the plugin and the :term:`broker`, publishing
+        messages to and receiving messages from it. Each plugin instance is associated
+        with exactly one backend.
+
+    broker
+        Responsible for receiving and publishing messages to all connected
+        :term:`backends <backend>`; All backends sharing the same broker will have
+        access to the same messages, allowing for inter-process communication. This is
+        typically handled by a separate entity like `Redis <https://redis.io/>`_
+
+    plugin
+        The :class:`ChannelsPlugin <.plugin.ChannelsPlugin>`, a central instance
+        managing :term:`subscribers <subscriber>`, reading messages from the
+        :term:`backend`, putting them in the appropriate :term:`event stream` and
+        publishing data to the backend
+
+
+Flowcharts
+++++++++++
+
+
+.. mermaid::
+    :align: center
+    :caption: Publishing flow from the application to the :term:`broker`
+
+    flowchart LR
+        Backend(Backend) --> Broker[(Broker)]
+
+        Plugin{{Plugin}} --> Backend
+
+        Application[[Application]] --> Plugin
+
+
+.. mermaid::
+    :align: center
+    :caption: Fanout flow of data from the :term:`broker` to the sockets, showing multiple plugin instances
+
+    flowchart TD
+        Broker[(Broker)]
+
+        Broker --> Backend_1(Backend)
+        Broker --> Backend_2(Backend)
+
+        Backend_1 --> Plugin_1{{Plugin}}
+        Backend_2 --> Plugin_2{{Plugin}}
+
+        Plugin_1 --> Subscriber_1[Subscriber]
+        Plugin_1 --> Subscriber_2[Subscriber]
+        Plugin_1 --> Subscriber_3[Subscriber]
+
+        Plugin_2 --> Subscriber_4[Subscriber]
+        Plugin_2 --> Subscriber_5[Subscriber]
+        Plugin_2 --> Subscriber_6[Subscriber]
+
+
+The ``ChannelsPlugin``
+----------------------
+
+.. currentmodule:: litestar.channels.plugin
+
+The :class:`ChannelsPlugin` acts as the central entity for managing channels and
+subscribers. It's used to publish messages, control how data is stored, manage
+subscribers, route handlers and configuration.
+
+
+.. tip::
+    The plugin makes itself available as a dependency under the ``channels`` key, which
+    means it's not necessary to import it and instead, it can be used from within route
+    handlers or other callables within the dependency tree directly
```

Comment:
```suggestion
    handlers or other callables directly.
```

Reply:
That's an important detail though. It does not automagically get injected into *any* callables.